### ISSUE TITLE
Typescript Svelte + Rollup incompatibility

### ISSUE THREAD

**Issue by joewagner** [tablelandnetwork/js-tableland#52]
## Expected Behavior
When using Svelte with the default bundler, [Rollup](https://rollupjs.org), This library should not cause warnings.

## Actual Behavior
Rollup complains with "The following packages did not export their `package.json`", and "`this` has been rewritten to `undefined`"

![Screen Shot 2022-05-27 at 9 53 52 AM](https://user-images.githubusercontent.com/1500995/170735472-c2874313-71f7-4a23-b8d1-7a76934deaf5.png)

## Steps to Reproduce the Problem

1. Follow the Svelte docs example using [degit](https://github.com/Rich-Harris/degit) and create a Svelte app from template that includes Typescript support
2. Import this SDK in the app
3. run the app with `npm run dev`

**Comment by joewagner** [comment]
Spent some time digging into this warning.  It comes down to Typescript's transpiler trying to support import/export and await/async in an es6 (or even es5) way while Rollup is trying to honor the rules of esm.

the Rollup docs give a simple explanation of what rollup is doing here: https://rollupjs.org/guide/en/#error-this-is-undefined 
TLDR; when using Javascript modules `this` is undefined at the top level of the file, and Rollup is trying to match the native behaviour so it's rewriting `this` to be undefined.
This codebase is transpiling the src Typsescript into es6 which means the top of everyfile gets `var __awaiter = (this && this.__awaiter) || function ...promise wrapper... `.  since there is a check for the existance of this at the top level Rollup is going to log a warning.

Things we could do:
 1. Nothing, it is just a warning.  Problem with this is that it's annoying and clutters the dev's console.
 2. Set the tsconfig for esm to target es2017.  I don't know of a specific problem here.  here's a compat table https://kangax.github.io/compat-table/es2016plus/ it seems that all of the platforms our users would need are supported, but I could easily be missing something here

cc: @carsonfarmer @awmuncy any thoughts on this?

**Comment by carsonfarmer** [comment]
Oh let's just go with #2. Worse case its too fresh for folks and they have to transpile it backward for their specific projects.

**Comment by joewagner** [comment]
I'm gonna close this since the newest version of Svelte and SvelteKit no longer have this issue.
